code/game/gamemodes/nuclear/nuclearbomb.dm

```diff
@@ -536,18 +536,19 @@ GLOBAL_VAR(bomb_set)
 /// round-end cinematic to play.
 /obj/machinery/nuclearbomb/proc/get_nuke_site()
 	var/turf/bomb_turf = get_turf(src)
-	if(bomb_turf)
-		if(!is_station_level(bomb_turf.z))
-			return NUKE_SITE_OFF_STATION_ZLEVEL
+	if(!bomb_turf)
+		return NUKE_SITE_INVALID
 
-		var/area/bomb_area = get_area(src)
-		for(var/A in GLOB.the_station_areas)
-			if(istype(bomb_area, A))
-				return NUKE_SITE_ON_STATION
+	if(!is_station_level(bomb_turf.z))
+		return NUKE_SITE_OFF_STATION_ZLEVEL
 
-		return NUKE_SITE_ON_STATION_ZLEVEL
+	var/area/bomb_area = get_area(src)
+	for(var/A in SSmapping.existing_station_areas)
+		if(istype(bomb_area, A))
+			return NUKE_SITE_ON_STATION
```

Comment:
I wonder if you could do
```suggestion
	if(get_area(src) in SSmapping.existing_station_areas)
		return NUKE_SITE_ON_STATION
```

Reply:
I don't see why not, let me test this in-game though

Reply:
Tested this in-game, seems fine